packages/esbuild/test/ts_project/BUILD.bazel

```diff
@@ -0,0 +1,42 @@
+load("//:index.bzl", "generated_file_test")
+load("//packages/esbuild/test:tests.bzl", "esbuild")
+load("//packages/typescript:index.bzl", "ts_project")
+
+exports_files(["tsconfig.json"])
+
+ts_project(
+    name = "main",
+    srcs = [":main.ts"],
+    declaration = True,
+    tsconfig = ":tsconfig.json",
+    deps = [
+        "//packages/esbuild/test/ts_project/module-dynamic",
+        "//packages/esbuild/test/ts_project/module-one",
+        "//packages/esbuild/test/ts_project/module-two",
+        "//packages/esbuild/test/ts_project/relative-module",
+    ],
+)
+
+esbuild(
+    name = "bundle",
+    args = [
+        "--keep-names",
+        "--resolve-extensions=.mjs,.js",
+    ],
+    entry_point = "main.ts",
+    format = "esm",
+    deps = [":main"],
+)
```

Comment:
Perhaps we should base this branch on 4.x, the API here has changed

Reply:
Done